ecosystem/typescript/sdk/src/plugins/coin_client.ts

```diff
@@ -54,8 +61,32 @@ export class CoinClient {
       // If this is the first time an account has received the specified coinType,
       // and this is set to false, the transaction would fail.
       createReceiverIfMissing?: boolean;
+      // fungible asset address if you wish to transfer a fungible asset
+      assetAddress?: MaybeHexString;
```

Comment:
```suggestion
      assetMetadataAddress?: MaybeHexString;
```

Let's call this `assetMetadataAddress` or something like that. Just calling this `assetAddress` is confusing as intuitively this sounds like the address of the actual asset we're transferring, but really what this argument is referring to is just the metadata object address.

Reply:
the asset is not always `Metadata`, it can also be `Object`, so naming it `Metadata` can be slightly wrong. It is mostly depends on the contract you interact with.

Reply:
then maybe we need to give this a different name then what I suggested, that being said that doesn't change the fact that the current name is even more confusing imo.

Reply:
maybe we can have `fungibleAssetAddress` since this is the fungible asset address, (i.e 0x123)